art/defences/trainer/adversarial_trainer_madry_pgd.py

```diff
@@ -54,8 +54,8 @@ class AdversarialTrainerMadryPGD(Trainer):
     def __init__(
         self,
         classifier: "CLASSIFIER_LOSS_GRADIENTS_TYPE",
-        nb_epochs: int = 391,
-        batch_size: int = 128,
+        nb_epochs: Optional[int] = 391,
```

Comment:
Although not part of the core change, is it worth adding a qualifier in the comments as to why 391 was chosen as the default argument? It's a non-standard number: is it based on the original paper? 

Should this (along with the batch size) have the Optional typing removed? If None is supplied here and then not explicitly passed in the fit it could cause a problem.

Reply:
That's correct, if both values for `nb_epochs` are None it will now raise a ValueError with an explanation. This is to remain backward compatible and to align all trainers to the same keyword arguments.

I think the 391 should actually be 205 epochs to represent 80'000 training steps on 50'000 CIFAR10 training samples with batch_size 128. We'll update it for ART 1.11 to keep the default behaviour for ART 1.10.x

Reply:
I opened issue #1644 for the default value update.